guides/v2.1/contributor-guide/backward-compatible-development/index.md

```diff
@@ -39,6 +39,12 @@ Mark the method with the `@deprecated` tag instead of removing it.
 
 Continue returning the same results from the method if possible, so the old functionality is preserved.
 
+If you need to remove the existing functionality and there is no alternative implementation, you MUST change the method code to
```

Comment:
Actually, you can ADD `trigger_error` method but don't replace the existing. The method should still return the same result.